### DISCUSSION TITLE
Ne pas afficher tarteaucitron dans une iframe / Do not display tarteaucitron in an iframe.

### DISCUSSION THREAD

**Discussion by lucmuller** [AmauriC/tarteaucitron.js#1061] [Q&A]
Dans l'un de mes sites, j'ai un système de modalbox qui permet, dans une page, d'incorporer une page du même site (même domaine) dans une iframe.

De fait, comme tarteaucitron est installé sur *toutes les pages* du site, si un nouvel utilisateur arrive sur le site et n'a pas fait de choix le bandeau TAC s'ouvre et dans la page principale et dans la page intégrée dans l'iframe.

Je ne crois pas que cela existe, mais je l'ai peut-être raté, est-ce qu'il y aurai un moyen de ne pas afficher TAC lorsque ce dernier est intégré dans une iframe pointant vers le même domaine ?

-------------
*lazy google translation*

In one of my sites, I have a modalbox system that allows, in a page, to embed a page from the same site (same domain) in an iframe.

In fact, as tarteaucitron is installed on *all the pages* of the site, if a new user arrives on the site and has not made a choice, the TAC banner opens both in the main page and in the page integrated into the 'iframe.

I don't believe this exists, but I may have missed it, will there be a way to not show TAC when it's embedded in an iframe pointing to the same domain?

**Comment by AmauriC** [comment]
Peut-être une piste du côté de `Content Security Policy` ?
https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP

**Reply by lucmuller** [reply]
Pas sur de comprendre ?
Je peux déjà déclarer quelque chose dans TAC pour le faire et/ou au niveau de mon iframe pour que ça soit fonctionnel ?

J'ai bien une solution technique pour empêcher d'avoir TAC pour les pages qui sont censés s'afficher dans l'iframe (elle sont placées à un endroit spécifique et ne sont jamais affiché en dehors de ce cadre).

Mais je me dis qu'un fonctionnement où TAC lui-même déduirai qu'il se trouve dans une iframe du même site et que donc il n'a pas besoin de s'afficher pourrai exister (?)

Après, peut-être que je me plante et qu'il faut conserver un usage générique : l'iframe affiche une page web. la page web contient TAC DONC TAC s'affiche et il n'y a aucune raison pour TAC de ne pas s'afficher.

Dans ce cas, c'est a impérativement à moi de faire l'adaptation technique dans mon site.

_C'est pour ça que j'ai ouvert le sujet dans la discussion et pas dans les issues_ ^^

**Reply by AmauriC** [reply]
Ok, ok je comprend mieux. Du côté de tac il est sûrement possible d'ajouter une option mais ça risque de complexifier encore davantage pour un besoin qui semble être de niche.
Si je me trompe et qu'il y a vrai besoin pour ce cas, aucun problème pour intégrer une option ;)

**Comment by nicozerr** [comment]
Tu peux aussi utiliser https://developer.mozilla.org/fr/docs/Web/API/Window/parent, et ne pas lancer TAC si window.parent != window.top

**Reply by lucmuller** [reply]
Yep, merci pour la suggestion.

Je n'ai pas vraiment de souci sur l'implémentation technique. Le point est d'ailleurs déjà traité de mon côté (j'ai carrément empêché que le script et les styles de TAC soient inclus dans ce lot de page).

Je voulais plutôt ouvrir la réflexion au niveau de TAC, voir de la gestion de bannière de cookies d'une manière générale sur la question :
Est-ce que l'on est censé afficher la bannière des cookies dans une iframe appartenant au site courant.

J'imagine que pour un site tiers, la question ne se pose pas... Quoi que je ne suis même pas allé vérifier la manière dont se comporte, par exemple, Google (genre maps ou youtube) lorsqu'il est embarqué dans une iframe. Est-ce qu'on a bien une demande d'acceptation des cookies ?

**Reply by AmauriC** [reply]
La demande pour Youtube fonctionne dans une iframe:
https://jsfiddle.net/ybosmdq0/
